plugins/ScheduledReports/API.php

```diff
@@ -454,6 +454,14 @@ public function generateReport($idReport, $date, $language = false, $outputType
 
                 $processedReport['segment'] = $segment;
 
+                if (!empty($processedReport['metadata']['imageGraphUrl']) && !empty($processedReport['metadata']['metrics'])) {
+                    $columns = '&columns=' . implode(',', array_keys($processedReport['metadata']['metrics']));
```

Comment:
@sgiehl any better way to achieve this ?
I tried setting columns in parameters here https://github.com/innocraft/plugin-CustomReports/blob/4.x-dev/GetCustomReport.php#L170

This sets the columns but the uniqueID logic breaks since we concat all the parameters and hence we get a empty report, I tried handling this by ignoring key with columns where we are setting uniqueID, it throws `Invalid API Module and/or API Action` exception

Reply:
@sgiehl Added this https://github.com/matomo-org/matomo/pull/18870/files to resolve above issues